src/fklearn/training/transformation.py

```diff
@@ -354,11 +354,8 @@ def apply_replacements(df: pd.DataFrame,
         Default value to replace when original value is not present in the `vec` dict for the feature
 
     """
-    def column_categorizer(col: str) -> np.ndarray:
-        replaced = df[col].map(vec[col])
-        unseen = df[col].notnull() & replaced.isnull()
-        replaced[unseen] = replace_unseen
-        return replaced
+    def column_categorizer(col: str) -> pd.Series:
+        return df[col].map(lambda x: vec[col].get(x, replace_unseen), na_action='ignore')
```

Comment:
is this faster than `.apply`? I don't know how map works under the hood, but if it implements a for loop in the backend for lambda function, than its just as bad as apply no?